src/api/routes/faucets.ts

```diff
@@ -230,9 +214,73 @@ export const FaucetRoutes: FastifyPluginAsync<
   const FAUCET_STACKING_WINDOW = 2 * 24 * 60 * 60 * 1000; // 2 days
   const FAUCET_STACKING_TRIGGER_COUNT = 1;
 
-  const STX_FAUCET_NETWORKS = () => getStxFaucetNetworks();
+  const STX_FAUCET_NETWORK = () => getStxFaucetNetwork();
   const STX_FAUCET_KEYS = (process.env.FAUCET_PRIVATE_KEY ?? testnetKeys[0].secretKey).split(',');
 
+  async function calculateSTXFaucetAmount(
+    network: StacksNetwork,
+    stacking: boolean
+  ): Promise<bigint> {
+    if (stacking) {
+      try {
+        const poxInfo = await clientFromNetwork(network).getPox();
+        let stxAmount = BigInt(poxInfo.min_amount_ustx);
+        const padPercent = new BigNumber(0.2);
+        const padAmount = new BigNumber(stxAmount.toString())
+          .times(padPercent)
+          .integerValue()
+          .toString();
+        stxAmount = stxAmount + BigInt(padAmount);
+        return stxAmount;
+      } catch (error) {
+        // ignore
+      }
+    }
+    return FAUCET_DEFAULT_STX_AMOUNT;
+  }
+
+  async function fetchNetworkChainID(network: StacksNetwork): Promise<number> {
+    const rpcClient = clientFromNetwork(network);
+    const info = await rpcClient.getInfo();
+    return info.network_id;
+  }
+
+  async function buildSTXFaucetTx(
+    recipient: string,
+    amount: bigint,
+    network: StacksNetwork,
+    senderKey: string,
+    nonce: bigint,
+    fee?: bigint
+  ): Promise<StacksTransaction> {
+    try {
+      const options: SignedTokenTransferOptions = {
+        recipient,
+        amount,
+        senderKey,
+        network,
+        memo: 'faucet',
+        anchorMode: AnchorMode.Any,
+        nonce,
+      };
+      if (fee) options.fee = fee;
+
+      // Detect possible custom network chain ID
+      network.chainId = await fetchNetworkChainID(network);
```

Comment:
@zone117x Just curious, why not use the `STACKS_CHAIN_ID` [env var](https://github.com/hirosystems/stacks-blockchain-api/blob/fix/faucet-keys/.env#L102C1-L102C16) here?